script/deploymentScript/DeployWorkflowSpecificFactory.s.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "forge-std/Script.sol";
+import "forge-std/console.sol";
+
+import {Restricted_PIM_Factory_v1} from
+    "src/factories/workflow-specific/Restricted_PIM_Factory_v1.sol";
+import {Immutable_PIM_Factory_v1} from
+    "src/factories/workflow-specific/Immutable_PIM_Factory_v1.sol";
+
+contract DeployWorkflowSpecificFactory is Script {
+    function run() public {
+        address orchestratorFactory =
+            vm.envAddress("ORCHESTRATOR_FACTORY_ADDRESS");
+        address trustedForwarder = vm.envAddress("TRUSTED_FORWARDER_ADDRESS");
+        string memory factoryType = vm.envString("FACTORY_TYPE");
+
+        vm.startBroadcast(vm.envUint("DEPLOYER_PRIVATE_KEY"));
+
+        if (isEqual(factoryType, "RESTRICTED")) {
+            {
+                console2.log(
+                    "Deploying Restricted_PIM_Factory_v1 at address: ",
+                    address(
+                        new Restricted_PIM_Factory_v1(
+                            orchestratorFactory, trustedForwarder
```

Comment:
I could be wrong, but this would also execute if both addresses (or at least one) is `address(0)`, right? Maybe we should add something like we have in the `DeploymentScript.s.sol` to ensure that the deployment is gonna be okay? (see [here](https://github.com/InverterNetwork/contracts/blob/2a8a4c80ff4f24a59546d4e6126b81bc51228c94/script/deploymentScript/DeploymentScript.s.sol#L458))

Reply:
Great idea!